src/Microsoft.FeatureManagement.AppInsightsTelemetryPublisher/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Microsoft.ApplicationInsights;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.FeatureManagement.Telemetry;
+
+namespace Microsoft.FeatureManagement.AppInsightsTelemetryPublisher
+{
+    /// <summary>
+    /// Extensions used to add feature management publisher functionality.
+    /// </summary>
+    public static class ServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds an event publisher that publishes feature evaluation events to Application Insights.
+        /// </summary>
+        /// <param name="services">The service collection that feature management services are added to.</param>
+        /// <returns>The <see cref="IServiceCollection"/> that was given as a parameter, with the publisher added.</returns>
+        public static IServiceCollection AddFeatureManagementTelemetryPublisherAppInsights(this IServiceCollection services)
+        {
+            //
+            // Add required services
+            services.AddSingleton<ITelemetryPublisher>(serviceProvider =>
```

Comment:
Would it be better to use TryAddSingleton()? 
If user implements a custom ITelemetry publisher and registers it
builder.Services.AddSingleton<ITelemetryPublisher>(new MyCustomTelemetryPublisher ());
What is the expected behavior after calling AddFeatureManagementTelemetryPublisherAppInsights?

Reply:
Good question. I think both of those options work. 

In my mind, this function does 1 thing, which is add the publisher. It makes sense for it to override, as if you call it, you expect something to happen. 

Although, I didn't think of the use case if they call it without registering a TelemetryClient first, which I could see happening as you're setting up the app.

Reply:
I opt for Try. It gives idempotence